hphp/hsl/src/str/transform_l.php

```diff
@@ -0,0 +1,314 @@
+)[]: string {
+  /* HH_FIXME[4390] missing [] */
+  return _Str\lowercase_l($string, $locale);
+}
+
+/**
+ * Returns the string padded to the total length (in characters) by appending
+ * the `$pad_string` to the left.
+ *
+ * If the length of the input string plus the pad string exceeds the total
+ * length, the pad string will be truncated. If the total length is less than or
+ * equal to the length of the input string, no padding will occur.
+ *
+ * To pad the string on the right, see `Str\pad_right_l()`.
+ * To pad the string to a fixed number of bytes, see `Str\pad_left()`.
+ */
+function pad_left_l(
+  Locale\Locale $locale,
+  string $string,
+  int $total_length,
+  string $pad_string = ' ',
+)[]: string {
+  /* HH_FIXME[4390] missing [] */
+  return _Str\pad_left_l($string, $total_length, $pad_string, $locale);
+}
+
+/**
+ * Returns the string padded to the total length (in characters) by appending
+ * the `$pad_string` to the right.
+ *
+ * If the length of the input string plus the pad string exceeds the total
+ * length, the pad string will be truncated. If the total length is less than or
+ * equal to the length of the input string, no padding will occur.
+ *
+ * To pad the string on the left, see `Str\pad_left()`.
+ * To pad the string to a fixed number of bytes, see `Str\pad_right()`
+ */
+function pad_right_l(
+  Locale\Locale $locale,
+  string $string,
+  int $total_length,
+  string $pad_string = ' ',
+)[]: string {
+  /* HH_FIXME[4390] missing [] */
+  return _Str\pad_right_l($string, $total_length, $pad_string, $locale);
+}
+
+/**
+ * Returns the "haystack" string with all occurrences of `$needle` replaced by
+ * `$replacement`.
+ *
+ * Strings will be normalized for comparison in encodings that support multiple
+ * representations, such as UTF-8.
+ *
+ * - For a case-insensitive search/replace, see `Str\replace_ci_l()`.
+ * - For multiple case-sensitive searches/replacements, see `Str\replace_every_l()`.
+ * - For multiple case-insensitive searches/replacements, see `Str\replace_every_ci_l()`.
+ */
+function replace_l(
+  Locale\Locale $locale,
+  string $haystack,
+  string $needle,
+  string $replacement,
+)[]: string {
+  /* HH_FIXME[4390] missing [] */
+  return _Str\replace_l($haystack, $needle, $replacement, $locale);
+}
+
+/**
+ * Returns the "haystack" string with all occurrences of `$needle` replaced by
+ * `$replacement` (case-insensitive).
+ *
+ * Locale-specific rules for case-insensitive comparisons will be used, and
+ * strings will be normalized before comparing if the locale specifies an
+ * encoding that supports multiple representations of the same characters, such
+ * as UTF-8.
+ *
+ * - For a case-sensitive search/replace, see `Str\replace_l()`.
+ * - For multiple case-sensitive searches/replacements, see `Str\replace_every_l()`.
+ * - For multiple case-insensitive searches/replacements, see `Str\replace_every_ci_l()`.
+ */
+// not pure: str_ireplace uses global locale for capitalization
```

Comment:
Is this comment about impurity accurate? We're passing in the Locale, so why would it use the global locale?